sema4ai/src/sema4ai_code/agent_cli.py

```diff
@@ -188,3 +192,70 @@ def create_agent_package(self, directory: str) -> ActionResult[str]:
             )
 
         return ActionResult(success=True, message=None)
+
+    @staticmethod
+    def _validate_agent_package(
+        directory: str, workspace: IWorkspace, monitor: IMonitor
+    ) -> tuple[bool, str]:
+        # [TODO] - uncomment this when the agent package validation is fixed
+        # Validate the agent package
+        # args = [
+        #     "package",
+        #     "metadata",
+        #     "--package",
+        #     directory,
+        # ]
+        # command_result = self._run_agent_cli_command(args)
+
+        # if not command_result.success:
+        #     return False, f"Error validating the agent package.\n{command_result.message}"
+
+        agent_spec = uris.from_fs_path(str(Path(directory) / "agent-spec.yaml"))
+        found = list(collect_agent_spec_diagnostics(workspace, agent_spec, monitor))
```

Comment:
Given that there are errors and warnings returned, you should filter just for errors (to skip warnings/hints).

Also, the error message just showing a full dict (`{found}`) is probably not ideal, rather it should return a formatted error message.

Reply:
Then we need to create a custom message that will show the line as well

Reply:
The new code isn't really filtering just for errors before deciding to stop the packaging.

Reply:
correct, my bad!